uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/calender-instance.js

```diff
@@ -0,0 +1,72 @@
+define([
+  'common/moment',
+  'leave-absences/shared/modules/models-instances',
+  'common/models/instances/instance',
+], function (moment, instances) {
+  'use strict';
+
+  instances.factory('CalenderInstance', [
+    'ModelInstance',
+    function (ModelInstance) {
+
+      /**
+       * This method checks whether a date matches the send type.
+       *
+       * @param {string} Date
+       * @param {string} Type of day
+       *
+       * @return {Boolean}
+       * Throws Error if date is not found in calenderData
+       */
+      function checkDate(date, dayType) {
+        var searchedDate = this.calenderData.find(function (data) {
+          return moment(data.date).isSame(moment(date));
+        });
+
+        if (!searchedDate) {
+          throw new Error("Date not found");
+        }
+
+        return searchedDate.type.name === dayType;
+      }
+
+      return ModelInstance.extend({
+
+        /**
+         * This object contains the calendar data.
+         * Default value is empty Array
+         */
+        calenderData: [],
```

Comment:
There is a method provided by the base `ModelInstance` service called `defaultCustomData()`, where you can define the default data. Please use that.

Also, i don't think `calendarData` is the best name for the property. The calendar stores its days no? So call it simply `days`, or `dates`